modules/_user_data/main.tf

```diff
@@ -1,3 +1,15 @@
+# The `cluster_service_cidr` is required when `create == true`
+# This is a hacky way to make that logic work, otherwise Terraform always wants a value
+# and supplying any old value like `""` or `null` is not valid and will silently
+# fail to join nodes to the cluster
+resource "null_resource" "validate_cluster_service_cidr" {
+  lifecycle {
+    precondition {
+      condition     = var.create ? length(local.cluster_service_cidr) > 6 : true # length(local.cluster_service_cidr) == 0
```

Comment:
Where does `6` come from?

Reply:
Theres probably a better way, but the gist of the check is that something needs to be provided when `create = true`

Reply:
Cleaned up the commented out code and added a note on the arbitrary value of 6 in https://github.com/terraform-aws-modules/terraform-aws-eks/pull/2963/commits/1f2ef2acfc0fe9762f16a9c47d0810b1996391b7

Reply:
Thanks for the explanation! Something like this is a bit more flexible (probably): `try(cidrhost(local.cluster_service_cidr, 0) != "::", false)`